File: 02 Programing Foundamations/02 SOFTWARE TECHNOLOGIES/Projects/C# Projects/C Blog/Blog/Models/Article.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace Blog.Models
{
    public class Article
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(50, ErrorMessage = "The {0} must be at least {2} characters long.", MinimumLength = 1)]
        public string Title { get; set; }

        [Required]
        [StringLength(250, ErrorMessage = "The {0} must be at least {2} characters long.", MinimumLength = 1)]
        public string Content { get; set; }

        [ForeignKey("Author")]
        public string AuthorId { get; set; }

        public virtual ApplicationUser Author { get; set; }

        public DateTime Date { get; set; }

        public Article()
        {
            this.Date = DateTime.Now;
        }

        public bool isAuthor(string name)
        {
            return this.Author.UserName.Equals(name);
        }         
    }
}

File: 02 Programing Foundamations/02 SOFTWARE TECHNOLOGIES/Projects/Java Projects/Java Full Blog/models/Article.js
const mongoose = require('mongoose');

let articleShema = mongoose.Schema({
    title:{type:String,required:true},
    content:{type:String, required:true},
    author:{type:mongoose.Schema.Types.ObjectId, required:true,ref:'User'},
    date:{type:Date,default:Date.now()},
    imagesPath:{type:String}
});

articleShema.method({
    trimContent:function () {
        return this.content=this.content.substr(0,100)+'...';
    }
});

const Article = mongoose.model('Article',articleShema);

module.exports = Article;